docs/docs/architecture.md

```diff
@@ -170,59 +175,62 @@ sequenceDiagram
-is to embed a TLS stack into the enclave and use something like HTTPS, but this technique has some problems
-and limitations that are resolved via [the Conclave Mail API](mail.md). Mail makes communication between enclaves and
-clients easy, as well as solving common problems faced with doing application development. 
+Clients must securely communicate with enclaves without the host spying on the messages. Conclave uses
+[Conclave Mail](mail.md) to send and receive encrypted messages between a client and an enclave. Conclave Mail makes 
+communication between enclaves and clients easier, more secure, and more efficient when compared to classical 
+architectures like HTTPS. 
 
 ### Testing and debugging
 
-Conclave provides full unit testing support for enclaves. Enclaves themselves can be compiled for one of four modes:
+Conclave provides full unit testing support for enclaves. You can compile enclaves in one of these four modes:
 
-* **Production/release**: fully encrypted and protected memory.
-* **Debug**: the same as production, but special instructions are provided that allow enclave memory to be read and modified.
-  This mode provides no protection but is otherwise a faithful recreation of the standard environment.
-* **Simulation**: SGX hardware isn't actually used at all. This is helpful during development when SGX capable hardware
-  may not be available but does still require a Linux environment or a Linux Docker container to run.
-* **Mock**: In mock mode the enclave class runs in the same JVM as the host, so interactions between the enclave
-  and host are all just regular function calls. You can step through using a debugger and enjoy the regular Java development
-  experience. See [here for more information on using and configuring mock mode](mockmode.md).
+* **Release**: Fully encrypted and protected memory used for the production version of an application.
+* **Debug**: It's the same as release mode, but it has special instructions that allow the enclave memory to be read 
+  and modified. This mode provides no protection but is a faithful recreation of the production environment that 
+  you can use for testing.
+* **Simulation**: This mode doesn't use SGX hardware at all. Simulation mode is helpful during development when 
+  SGX-capable hardware may not be available. You need a Linux environment or a Linux Docker container to run 
+  enclaves in simulation mode.
+* **Mock**: In mock mode, the enclave class runs in the same JVM as the host. So, interactions between the enclave
+  and the host are regular function calls. You can step through using a debugger and enjoy a regular Java development
+  experience. See [here](mockmode.md) for more information on using and configuring mock mode.
 
-The modes must match between how the enclave was compiled and how it's loaded. This is handled for you automatically.
+The modes must match between how the enclave was compiled and how it's loaded. Conclave automatically handles this for
```

Comment:
I'm not sure what the first two sentences are talking about. Delete them, keep the last one.

Reply:
Deleted.